snapcraft/commands/lint.py

```diff
@@ -0,0 +1,113 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2023 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Snapcraft lint commands."""
+
+import argparse
+import os
+import textwrap
+from pathlib import Path
+from typing import Optional
+
+from craft_cli import BaseCommand, emit
+from craft_cli.errors import ArgumentParsingError
+from overrides import overrides
+
+
+class LintCommand(BaseCommand):
+    """Lint-related commands."""
+
+    name = "lint"
+    help_msg = "Lint a snap file"
+    overview = textwrap.dedent(
+        """
+        Lint an existing snap file.
+
+        The snap is installed and linted inside a build environment. If an assertion
+        file exists in the same directory as the snap file with the name
+        `<snap-name>.assert`, it will be used to install the snap in the instance.
+        """
+    )
+
+    @overrides
+    def fill_parser(self, parser: "argparse.ArgumentParser") -> None:
+        parser.add_argument(
+            "snap_file",
+            metavar="snap-file",
+            type=Path,
+            help="Snap file to lint",
+        )
+        parser.add_argument(
```

Comment:
isn´t LXD the default? I believe we have this in other commands for backwards compatibility only